### PR TITLE
feat: implement 0x36 - CALLDATASIZE Opcode

### PR DESCRIPTION
**Pull request by TAdev0**
**Pull Request type**

 Feature : implementation of CALLDATASIZE opcode
 
**What is the current behavior?**
Resolves: https://github.com/kkrt-labs/kakarot-ssj/issues/122

**What is the new behavior?**

none

**Does this introduce a breaking change?**

No

### CODE REVIEW

File: crates/evm/src/instructions/environmental_information.cairo [hunk 1]
```diff
@@ -57,6 +57,11 @@ impl EnvironmentInformationImpl of EnvironmentInformationTrait {
     /// Get the size of return data.
     /// # Specification: https://www.evm.codes/#36?fork=shanghai
     fn exec_calldatasize(ref self: ExecutionContext) -> Result<(), EVMError> {
+        let call_data: Span<u8> = self.call_context().call_data();
+        let call_data_size: u32 = call_data.len();
+        let result: u256 = call_data_size.into();
+
+        self.stack.push(result)?;
         Result::Ok(())
```
Comment (lambda-0x) [inline crates/evm/src/instructions/environmental_information.cairo:65]:
```suggestion
        self.stack.push(result)
```

Comment (TAdev0) [reply crates/evm/src/instructions/environmental_information.cairo]: thank you for the review!

File: crates/evm/src/instructions/environmental_information.cairo [hunk 2]
```diff
@@ -57,7 +57,11 @@ impl EnvironmentInformationImpl of EnvironmentInformationTrait {
     /// Get the size of return data.
     /// # Specification: https://www.evm.codes/#36?fork=shanghai
     fn exec_calldatasize(ref self: ExecutionContext) -> Result<(), EVMError> {
-        Result::Ok(())
+        let call_data: Span<u8> = self.call_context().call_data();
+        let call_data_size: u32 = call_data.len();
+        let result: u256 = call_data_size.into();
```
Comment (enitrat) [inline crates/evm/src/instructions/environmental_information.cairo:62]: can you make it into a one liner instead?

File: crates/evm/src/tests/test_instructions/test_environment_information.cairo [hunk 3]
```diff
@@ -3,7 +3,11 @@ use evm::tests::test_utils::{setup_execution_context, evm_address, callvalue};
 use evm::stack::StackTrait;
 use option::OptionTrait;
 use starknet::EthAddressIntoFelt252;
+use evm::context::ExecutionContext;
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_environment_information.cairo:6]: you can group these imports together

`use evm::context::{ExecutionContext, ExecutionContextTrait, CallContextTrait, CallContextTrait};`

File: crates/evm/src/tests/test_instructions/test_environment_information.cairo [hunk 4]
```diff
@@ -41,3 +45,19 @@ fn test__exec_callvalue() {
...
     assert(ctx.stack.pop().unwrap() == callvalue(), 'should be `123456789');
 }
+
+#[test]
+#[available_gas(20000000)]
+fn test_calldata_size() {
+    // Given
+    let mut ctx = setup_execution_context();
+    let call_data: Span<u8> = ctx.call_context().call_data();
+    let call_data_len = call_data.len();
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_environment_information.cairo:55]: unused var

**Comment by TAdev0** [comment]
@enitrat  now it should be good !

File: crates/evm/src/tests/test_instructions/test_environment_information.cairo (hunk 4 above)
Comment (TAdev0) [reply crates/evm/src/tests/test_instructions/test_environment_information.cairo]: oh my bad... thank you.

**Review by enitrat — APPROVED** [review]
lgtm!

**Pull request merged**